components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,634 @@
+---
+-- @Liquipedia
+-- wiki=clashroyale
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local DateExt = require('Module:Date/Ext')
+local FnUtil = require('Module:FnUtil')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Page = require('Module:Page')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local CardNames = mw.loadData('Module:CardNames')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input')
+local Streams = Lua.import('Module:Links/Stream')
+
+local OpponentLibrary = require('Module:OpponentLibraries')
+local Opponent = OpponentLibrary.Opponent
+
+local UNKNOWN_REASON_LOSS_STATUS = 'L'
+local DEFAULT_WIN_STATUS = 'W'
+local DEFAULT_WIN_RESULTTYPE = 'default'
+local NO_SCORE = -1
+local SCORE_STATUS = 'S'
+local ALLOWED_STATUSES = {DEFAULT_WIN_STATUS, 'FF', 'DQ', UNKNOWN_REASON_LOSS_STATUS}
+local MAX_NUM_OPPONENTS = 2
+local DEFAULT_BEST_OF = 99
+local NOW = os.time(os.date('!*t') --[[@as osdateparam]])
+local ROYALE_API_PREFIX = 'https://royaleapi.com/'
+local MAX_NUM_MAPS = 30
+
+-- containers for process helper functions
+local matchFunctions = {}
+local walkoverProcessing = {}
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+---@param match table
+---@return table
+function CustomMatchGroupInput.processMatch(match)
+	Table.mergeInto(match, MatchGroupInput.readDate(match.date))
+	match = matchFunctions.getExtraData(match)
+	match = CustomMatchGroupInput._getTournamentVars(match)
+	match = CustomMatchGroupInput._adjustData(match)
+	match = matchFunctions.getVodStuff(match)
+
+	return match
+end
+
+---@param template string
+---@return string|nil icon
+---@return string|nil iconDark
+function CustomMatchGroupInput.getIcon(template)
+	local raw = mw.ext.TeamTemplate.raw(template)
+	if raw then
+		local icon = Logic.emptyOr(raw.image, raw.legacyimage)
+		local iconDark = Logic.emptyOr(raw.imagedark, raw.legacyimagedark)
+		return icon, iconDark
+	end
+end
+
+---@param obj table
+---@param scores table
+function walkoverProcessing.walkover(obj, scores)
+	if #scores ~= 2 then -- since we always have 2 opponents
+		error('Unexpected number of opponents when calculating winner')
+	elseif Array.all(scores, function(score)
+			return Table.includes(ALLOWED_STATUSES, score) and score ~= DEFAULT_WIN_STATUS
+		end) then
```

Comment:
It looks like this function is always called, so if we have a numeric score we would set double walkover?